docs/proposals/default-service-plans.md

```diff
@@ -0,0 +1,657 @@
+upstream Helm charts do today.
+
+So they add a ServiceInstance to their chart, and when they get to the class and
+plan fields in the spec, they hit a wall. The author doesn't know the
+capabilities of the cluster on which their chart will be deployed and aren't in
+a position to provide the class and plan. Neither do they know the secret keys,
+as it's broker specific. The author is forced to add required arguments to their
+chart, passing the buck on figuring out the right class, plan, instance
+parameters and secret keys to the person deploying the chart.
+
+Now consider a user who wants to get up and running quickly, so naturally they
+chose to use a "Service Catalog Ready"* Helm chart. They don't know or care
+about the details, and want to get the application deployed with sane defaults
+so that they can try it out. Before they can even install the application they
+need to:
+
+* Realize that they need to specify a class and plan.
+* Figure out what a broker is and which one is installed on their cluster.
+* Hunt down a list of suitable classes and plans.
+* Decide which class and plan is right for a trying out the application.
+* Fill in the class and plan, install the chart only to later learn that there
+    is a set of "magic" parameters that they should have specified when
+    provisioning the instance.
+* Scour the internet looking for undocumented, totally optional but it won't
+    work without them, parameters for their broker because brokers aren't
+    required to provide a schema.
+* Blindly copy/paste some parameters they found in a gist somewhere.
+* Realize that the broker is returning MYSQL_PASSWORD, while the app expects
+    DB_PASSWORD, so they update the pod spec with that and redeploy the app.
+
+**This is not the user experience that we want for Service Catalog.**
+
+\* _I am trying to make "Service Catalog Ready" charts and "Service Catalog Enabled" clusters a thing._ 😇
+
+This is a solved problem in Kubernetes for volumes: storage classes. They
+encapsulate provisioner-specific details and allow users to either select one
+using a friendly name like "slow", or omit the class and use the default set by
+the operator. Service Catalog is very close to this model today, and I am
+proposing that we "fill in the gaps" following the precedent set by storage.
+
+## Goals and Non-Goals
+
+* Enable using the same manifest across cloud providers and brokers.
+* Reduce the burden of using Service Catalog by having the brokers and cluster
+  operators to define defaults.
+* Non-invasively add optional fields to existing resources to support this
+  feature. This should not interfere with existing functionality.
+* Avoid making changes to Service Catalog that aren't _strictly_ required to
+  support this feature. For example, we aren't going to rename Service Bindings
+  to Service Claims.
+* Dynamic provisioning concepts is out of scope. Concepts such as binding to
```

Comment:
wording here confuses me a bit, dynamic provisioning is discussed below in Service Plan... is the idea that we're only discussing the creation of resources in this proposal?